mono/utils/mono-compiler.h

```diff
@@ -179,22 +179,42 @@ typedef int32_t __mono_off32_t;
 #endif
 
 #if !defined(mmap)
+
+#ifdef __cplusplus
+extern "C" {
```

Comment:
Same line? Binding for this PR? Understood though.

Reply:
I assume that since you are touching that, it is necessary. So, yes, binded to this PR, but for any other symbol you will need **going forward** marked as `extern "C"`, use `G_EXTERN_C` or `MONO_EXTERN_C` appropriately and not `G_BEGIN_DECLS`/`G_END_DECLS`.

Reply:
The necessity here is a little unclear, due to all the ifdefs and the ifdefs are wierd as well -- ifdef against function names isn't generally correct, but people often think it is.

The original intent is clear though and adding extern "C" (by whatever name you chose) here honors that intent. I will add `MONO_EXTERN_C`.

Reply:
I am reluctant to have `mono-compiler.h` depend on `mono-publib.h`.
That is probably why I used plain `extern "C"` here in the first place.

For example:
```
libgc/darwin_stop_world.c:#include "mono/utils/mono-compiler.h"
libgc/pthread_stop_world.c:#include "mono/utils/mono-compiler.h"
libgc/pthread_support.c:#include "mono/utils/mono-compiler.h"
```

Reply:
Maybe duplicate the `#define`s?

Reply:
I think so. The grepabability of same-line EXTERN_C is valuable.